File: CapstoneProjectAPI/Models/DTOs/UserDetailsResponseDto.cs
namespace CapstoneProjectAPI.Models.DTOs
{
    public class UserDetailsResponseDto
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public bool IsAdmin { get; set; }
        public int DepartmentId { get; set; }
        public string DepartmentName { get; set; } = string.Empty;
        public int? ManagerId { get; set; }
        public string? ManagerName { get; set; }
        public int Level { get; set; }
        public bool IsActive { get; set; }
    }
}

File: CapstoneProjectFrontend/src/app/dtos/user-details-response.dto.ts
import { PaginatedResponse } from "../helpers";

export interface UserDetails {
    id: number;
    name: string;
    email: string;
    isAdmin: boolean;
    departmentId: number;
    departmentName: string;
    managerId: number | null;
    managerName: string | null;
    level: number;
    isActive: boolean
}
export type UserDetailsResponseDto = PaginatedResponse<UserDetails>;